Content.Server/Fluids/EntitySystems/AbsorbentSystem.cs

```diff
@@ -117,30 +117,43 @@ private bool TryTransferAbsorber(EntityUid user, EntityUid used, EntityUid targe
         if (!_solutionSystem.TryGetRefillableSolution(target, out var refillableSolution, refillable: refillable))
             return false;
 
-        if (refillableSolution.Volume <= 0)
+        if (refillableSolution.Volume < 0)
         {
             var msg = Loc.GetString("mopping-system-target-container-empty", ("target", target));
             _popups.PopupEntity(msg, user, user);
             return false;
         }
 
+        var transferAmount = component.PickupAmount / PuddleSystem.EvaporationReagentRatio -
+            absorberSoln.GetReagentQuantity(PuddleSystem.EvaporationReagent);
+
+        transferAmount = transferAmount > 0 ? transferAmount : 0;
+
         // Remove the non-water reagents.
         // Remove water on target
         // Then do the transfer.
-        var nonWater = absorberSoln.SplitSolutionWithout(component.PickupAmount, PuddleSystem.EvaporationReagent);
```

Comment:
I needed to delay the movement of nonWater since now that does make a difference: I remove water from the container first so that I can just query `refillableSolution.AvailableVolume` later to check how much space there is for adding dirt volume to the container.